lib/astarte_data_access/config.ex

```diff
@@ -38,4 +57,162 @@ defmodule Astarte.DataAccess.Config do
     os_env: "CASSANDRA_NODES",
     type: CQExNodes,
     default: [{"localhost", 9042}]
+
+  @envdoc """
+  The Xandra authenticator that implements support for PasswordAuthenticator to authenticate
+  with Cassandra server.
+  """
+  app_env :xandra_authentication_module, :astarte_data_access, :xandra_authentication_module,
+    os_env: "XANDRA_AUTHENTICATION_MODULE",
+    type: :module,
+    default: Xandra.Authenticator.Password
+
+  @envdoc """
+  The CQEx authenticator that implements support for PasswordAuthenticator to authenticate
+  with Cassandra server.
+  """
+  app_env :cqex_authentication_module, :astarte_data_access, :cqex_authentication_module,
+    os_env: "CQEX_AUTHENTICATION_MODULE",
+    type: :module,
+    default: :cqerl_auth_plain_handler
+
+  @envdoc "The username used to log into cassandra. Defaults to 'cassandra'."
+  app_env :cassandra_username, :astarte_data_access, :cassandra_username,
+    os_env: "CASSANDRA_USERNAME",
+    type: :binary,
+    default: "cassandra"
+
+  @envdoc "The password used to log into cassandra. Defaults to 'cassandra'."
+  app_env :cassandra_password, :astarte_data_access, :cassandra_password,
+    os_env: "CASSANDRA_PASSWORD",
+    type: :binary,
+    default: "cassandra"
+
+  @envdoc """
+  Whether or not results of and parameters to Xandea.execute/4 will have atom keys.
+  Defaults to false.
+  """
+  app_env :atom_keys, :astarte_data_access, :atom_keys,
+    os_env: "CASSANDRA_ATOM_KEYS_ENABLED",
+    type: :boolean,
+    default: "false"
+
+  @envdoc "The number of connections to start for the pool, Defaults to 1."
+  app_env :pool_size, :astarte_data_access, :pool_size,
+    os_env: "CASSANDRA_POOL_SIZE",
+    type: :integer,
+    default: 1
+
+  @envdoc "Enable SSL for Cassandra connections. Defaults to false."
+  app_env :ssl_enabled, :astarte_data_access, :ssl_enabled,
+    os_env: "CASSANDRA_SSL_ENABLED",
+    type: :boolean,
+    default: false
+
+  @envdoc "Disable Server Name Indication. Defaults to false."
+  app_env :ssl_disable_sni, :astarte_data_access, :ssl_disable_sni,
+    os_env: "CASSANDRA_SSL_DISABLE_SNI",
+    type: :boolean,
+    default: false
+
+  @envdoc """
+  Specify the hostname to be used in TLS Server Name Indication extension.
+  If not specified, the cassandra nodes will be used. This value is used
+  only if Server Name Indication is enabled.
+  """
+  app_env :ssl_custom_sni, :astarte_data_access, :ssl_custom_sni,
+    os_env: "CASSANDRA_SSL_CUSTOM_SNI",
+    type: :binary
+
+  @envdoc """
+  Specifies the certificates of the root Certificate Authorities to be trusted.
+  When not specified, the bundled cURL certificate bundle will be used.
+  """
+  app_env :ssl_ca_file, :astarte_data_access, :ssl_ca_file,
+    os_env: "CASSANDRA_SSL_CA_FILE",
+    type: :binary
```

Comment:
You can probably pass `Castore.ca_cert_file()` as default here and you can avoid manually handling it